audb/core/load.py

```diff
@@ -933,35 +1026,32 @@ def load_header(
             to the database header before storing it in cache
         overwrite: always load header from backend
             and overwrite the one found in ``db_root``
+        timeout: maximum wait time if another thread or process is already
+            accessing the database. If timeout is reached, ``None`` is
+            returned. If timeout < 0 the method will block until the
+            database can be accessed
 
     Returns:
         database header and backend
 
     """
-    backend = None
-    local_header = os.path.join(db_root, define.HEADER_FILE)
-    if overwrite or not os.path.exists(local_header):
-        backend = lookup_backend(name, version)
-        remote_header = backend.join(name, define.HEADER_FILE)
-        if add_audb_meta:
-            db_root_tmp = database_tmp_folder(db_root)
-            local_header = os.path.join(db_root_tmp, define.HEADER_FILE)
-        backend.get_file(remote_header, local_header, version)
-        if add_audb_meta:
-            db = audformat.Database.load(db_root_tmp, load_data=False)
-            db.meta['audb'] = {
-                'root': db_root,
-                'version': version,
-                'flavor': flavor.arguments,
-                'complete': False,
-            }
-            db.save(db_root_tmp, header_only=True)
-            audeer.move_file(
-                os.path.join(db_root_tmp, define.HEADER_FILE),
-                os.path.join(db_root, define.HEADER_FILE),
+    db_lock_path = database_lock_path(db_root)
+    db, backend = None, None
+
+    try:
+        with filelock.FileLock(db_lock_path, timeout=timeout):
+            db, backend = _load_header(
+                db_root,
+                name,
+                version,
+                flavor,
+                add_audb_meta,
+                overwrite,
             )
-            audeer.rmdir(db_root_tmp)
-    return audformat.Database.load(db_root, load_data=False), backend
+    except filelock.Timeout:
+        pass
```

Comment:
Should we maybe show a message that a timeout occurred when `verbose = True`?
Or is this done by `filelock.Timeout` anyway?

Reply:
No, it does not automatically show a message. But we can do that of course.

Reply:
If timeout is exceeded it now shows:

```
UserWarning: Lock could not be acquired. Timeout exceeded.
```

It does not depend on `verbose`, otherwise we have to introduce a `verbose` argument to `info.header()` first. I also don't think it's necessary since warnings can be omitted by setting a warning filter.

Reply:
Sounds reasonable to not handle it with `verbose`.